index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
The `Semigroup` here is `Maybe a`, so if someone attempts to `concat` with anything that is not `Maybe a` (the `a` must be the same as well), then it is unspecified what happens. There is nothing specifying that the `a` is also a `Semigroup` in the implementation, but I'm not sure what you'd do there either. 

How should that case be handled? It will already throw a runtime error if `concat` doesn't exist, what else could you do? More descriptive error possibly?